nautobot_ssot/integrations/servicenow/diffsync/models.py

```diff
@@ -29,17 +29,18 @@ def map_data_to_sn_record(self, data, mapping_entry, existing_record=None):
                 sys_id = None
                 if "column" not in mapping["reference"]:
                     raise NotImplementedError
-                column_name = mapping["reference"]["column"]
+                # column_name = mapping["reference"]["column"]
                 if value is not None:
+                    # Cacheing disabled as it was introducing instability and unpredictable results during delete method testing
```

Comment:
Did you figure out why this was the case? Can we possibly fix the cache mechanism instead? I reckon its existence is probably necessary for some bigger deployments. Also if we do decide to omit caching, can you delete the commented out code?

Reply:
I figured out what the issue with the caching functionality was and handled it in the new commits. Please review the newest commits for the changes.